test/azul_test_case.py

```diff
@@ -1,9 +1,50 @@
-import os
-from unittest import TestCase, mock
+from unittest import TestCase
 
 import boto3.session
 
 
+class AlwaysTearDownTestCase(TestCase):
+    """
+    SafeTestCase makes sure that tearDown / cleanup methods are always run when
+    they should be.
+
+    This means that
+
+    - if a KeyboardInterrupt is raised in a test, then tearDown, tearDownClass,
+      and tearDownModule will all still run
+    - if any exception is raised in a setUp, then tearDown, tearDownClass,
+      and tearDownModule will all still run
+
+    Caveats:
+
+    - All tearDown methods should pass even if their corresponding setUps don't
+      run at all, as in the case of a KeyboardInterrupt or other exception.
+    - If an exception is raised in SetUpClass or SetUpModule, tearDowns will
+      not be run. (Working around this involves making a custom TestSuite which
+      will be ignored if you run tests through PyCharm)
+    """
+
+    def run(self, result=None):
+        test_method = getattr(self, self._testMethodName)
+        wrapped_test = self._cleanup_wrapper(test_method, KeyboardInterrupt)
+        setattr(self, self._testMethodName, wrapped_test)
+
+        self.setUp = self._cleanup_wrapper(self.setUp, BaseException)
+
+        return super().run(result)
+
+    def _cleanup_wrapper(self, method, exception):
```

Comment:
`exception` indicates an instance. But this is a class. So
```suggestion
    def _cleanup_wrapper(self, method, exception_cls):
```